library/test/src/types.rs

```diff
@@ -140,6 +142,28 @@ impl TestDesc {
             }
         }
     }
+
+    pub fn test_mode_string(&self) -> String {
```

Comment:
Nit: Since none of these are dynamic, I'd just go for `&'static str` and not allocate anything. (Saves a bunch of `.to_string()` calls.)